src/main/g8/scalafix/build.sbt

```diff
@@ -14,41 +18,93 @@ inThisBuild(
         url("https://example.com")
       )
     ),
-    addCompilerPlugin(scalafixSemanticdb),
-    scalacOptions ++= List(
-      "-Yrangepos",
-      "-P:semanticdb:synthetics:on"
-    )
+    semanticdbEnabled := true,
+    semanticdbVersion := scalafixSemanticdb.revision
   )
 )
 
-publish / skip := true
+lazy val root = (project in file("."))
+  .aggregate(
+    rules.projectRefs ++
+      input.projectRefs ++
+      output.projectRefs ++
+      tests.projectRefs: _*
+  )
+  .settings(
+    publish / skip := true
+  )
 
-lazy val rules = project.settings(
-  moduleName := "scalafix",
-  libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
-)
+lazy val rules = projectMatrix
+  .settings(
+    moduleName := "scalafix",
+    libraryDependencies += "ch.epfl.scala" %% "scalafix-core" % V.scalafixVersion
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(rulesCrossVersions)
 
-lazy val input = project.settings(
-  publish / skip := true
-)
+lazy val input = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .jvmPlatform(
+    scalaVersions = rulesCrossVersions,
+    settings = Seq(
+      scalacOptions += "-P:semanticdb:synthetics:on"
+    )
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val output = project.settings(
-  publish / skip := true
-)
+lazy val output = projectMatrix
+  .settings(
+    publish / skip := true
+  )
+  .defaultAxes(VirtualAxis.jvm)
+  .jvmPlatform(scalaVersions = rulesCrossVersions :+ scala3Version)
 
-lazy val tests = project
+lazy val testsAggregate = Project("tests", file("tests/aggregate"))
+  .aggregate(tests.projectRefs: _*)
```

Comment:
added to maintain backward compatibility on CI and on [docs](https://scalacenter.github.io/scalafix/docs/developers/setup.html)